app/api/auth.py

```diff
@@ -0,0 +1,182 @@
+from datetime import timedelta
+from app import db, jwt
+from flask import Blueprint, request, jsonify
+from flask_jwt_extended import create_access_token, jwt_required, get_jwt
+from app.models import Users, Account, TokenBlockList
+
+bp = Blueprint('auth', __name__)
+
+
+@bp.route('/signup', methods=['POST'])
+def signup():
+    try:
+        data = request.get_json()
+
+        attributes = data.get('data', {}).get('attributes', {})
+        username = attributes.get('username')
+        password = attributes.get('password')
+        account_name = attributes.get('account_name')
+
+        if not attributes or not username or not password or not account_name:
+            return jsonify({'errors': [{
+                'status': '422',
+                'detail': 'Some attributes are missing. '
+                'Username, password and account name are required'
+            }]}), 422
+
+        existing_user = Users.query.filter_by(username=username).first()
+        if existing_user:
+            return jsonify({'errors': [{
+                'status': '422', 
+                'detail': 'User already exists'
+            }]}), 422
```

Comment:
All errors have the same structure,  it would be easier and protect you from errors if you defined error structure in TypedDict and always use it, e.g.:
```
class JsonApiError(typing.TypedDict):
    status: str
    detail: str
 
class JsonApiErrorResponse(typing.TypedDict):
    errors: list[JsonApiError]
....
return jsonify(JsonApiErrorResponse(errors=[JsonApiError(status="422...
```